src/NodaTime/Instant.cs

```diff
@@ -655,6 +655,28 @@ public static Instant FromUnixTimeTicks(long ticks)
         public long ToUnixTimeSeconds() =>
             duration.FloorDays * (long) SecondsPerDay + duration.NanosecondOfFloorDay / NanosecondsPerSecond;
 
+        /// <summary>
+        /// Gets the number of seconds since the Unix epoch, along with remaining nanoseconds.
+        /// Negative values represent instants before the Unix epoch.
+        /// </summary>
+        /// <remarks>
+        /// If the number of nanoseconds in this instant is not an exact number of seconds
+        /// the value is truncated towards the start of time, 
+        /// </remarks>
+        /// <value>The number of seconds since the Unix epoch.</value>
+        [Pure]
+        [TestExemption(TestExemptionCategory.ConversionName)]
+        public long ToUnixTimeSecondsAndNanoseconds(out int nanoseconds)
+        {
+            // These look like unrelated operations, but ToUnixTimeSeconds uses
+            // duration.NanosecondOfFloorDay / NanosecondsPerSecond. We could potentially
+            // use DivRem here instead, but it feels slightly cleaner this way,
```

Comment:
I don't think so - because we'd be using DivRem of duration.NanosecondOfFloorDay, which is always non-negative. (And then adding in the FloorDay part as well).

Happy to try that as a separate commit, so you can compare and contrast?

Reply:
Ah, I was originally thinking of computing a total ns-since-epoch (which can be negative) and then doing divrem with `NanosecondsPerSecond`, but in any case it now occurs to me that the dividend probably won't fit into an int64 to start with.

Yeah, maybe let's see whether the version with DivRem is clearer than what we have here?

Reply:
Done in another commit.